### PR TITLE
Improve "Defaults to" by putting to end of arg in docstring and ensuring backticks have proper spacing

### PR DESCRIPTION
**Pull request by SamuelMarks**
If you prefer I can change all arguments to be how the popular Sphinx documentation generator parser works: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

EDIT - Actually created a separate issue to discuss this #18944

Made much smaller and fewer changes than my previous (similar) PRs to improve chance of merging without controversy :)

### CODE REVIEW

File: keras/layers/layer.py [hunk 1]
```diff
@@ -459,7 +460,7 @@ def add_weight(
         Args:
             shape: Shape tuple for the variable.
                 Must be fully-defined (no `None` entries).
-                Defaults to `()` (scalar) if unspecified.
+                (scalar) if unspecified. Defaults to `()`.
```
Comment (fchollet) [inline keras/layers/layer.py:463]: The "(scalar) if unspecified" should go afterwards

File: keras/ops/nn.py [hunk 2]
```diff
@@ -1280,8 +1280,8 @@ def one_hot(x, num_classes, axis=-1, dtype=None):
         x: Integer tensor to be encoded. The shape can be
             arbitrary, but the dtype should be integer.
         num_classes: Number of classes for the one-hot encoding.
-        axis: Axis along which the encoding is performed. Defaults to
-            `-1`, which represents the last axis.
+        axis: Axis along which the encoding is performed.
+            `-1` represents the last axis. Defaults to `-1`.
```
Comment (fchollet) [inline keras/ops/nn.py:1284]: Again, "which represents..." should go afterwards.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras/ops/nn.py (hunk 2 above)
Comment (SamuelMarks) [reply keras/ops/nn.py:1284]: I will move it if you request again, however there is ambiguity if "defaults to" isn't the final part of the argument. E.g., see https://github.com/keras-team/keras/blob/b6618f7/keras/utils/dataset_utils.py#L24-L28. Although there is no concrete specification of the Google doc format, in Sphinx they put "Defaults to" at the end of the argument: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

**Comment by sachinprasadhs** [comment]
Please rebase the code to follow the latest code structure /keras/src/..

**Comment by SamuelMarks** [comment]
@sachinprasadhs No problem. Merged and resolved conflicts.

**Review by fchollet — APPROVED** [review]
LGTM, thanks!

**Pull request merged**